savant/utils/artist/artist_gpumat.py

```diff
@@ -180,6 +181,77 @@ def add_bbox(
                 bg_color=bg_color,
             )
 
+    def add_rounded_rect(
+        self,
+        bbox: BBox,
+        radius: int,
+        bg_color: Tuple[float, float, float],  # BGR
+    ):
+        self.__init_overlay()
+
+        cv2.rectangle(
+            self.overlay,
+            (int(bbox.left), int(bbox.top + radius)),
+            (int(bbox.right), int(bbox.bottom - radius)),
+            convert_color(bg_color),
+            -1
+        )
+        cv2.rectangle(
+            self.overlay,
+            (int(bbox.left + radius), int(bbox.top)),
+            (int(bbox.right - radius), int(bbox.bottom)),
+            convert_color(bg_color),
+            -1
+        )
+        # top-left
+        cv2.ellipse(
```

Comment:
I think it would be more readable to draw ellipses in a for-loop (iterate over center and angles).